mrmustard/lab_dev/transformations/base.py

```diff
@@ -42,25 +42,69 @@ class Transformation(CircuitComponent):
 
     def inverse(self) -> Transformation:
         r"""Returns the inverse of the transformation."""
+        if len(self.wires.input) != len(self.wires.output):
+            raise NotImplementedError(
+                "Only Transformations with the same number of input and output wires are supported."
+            )
         if not isinstance(self.representation, Bargmann):
             raise NotImplementedError("Only Bargmann representation is supported.")
         if self.representation.ansatz.batch_size > 1:
             raise NotImplementedError("Batched transformations are not supported.")
+
+        # compute the inverse
         A, b, _ = self.dual.representation.conj().triple  # apply X
-        almost_inverse = self.__class__.from_bargmann(
-            [0], (math.inv(A[0]), -math.inv(A[0]) @ b[0], 1 + 0j)
+        almost_inverse = self._from_attributes(
+            Bargmann(math.inv(A[0]), -math.inv(A[0]) @ b[0], 1 + 0j),
+            self.wires,
+            "",
         )
         almost_identity = (
             self >> almost_inverse
         )  # TODO: this is not efficient, need to get c from formula
         invert_this_c = almost_identity.representation.c
-        actual_inverse = self.__class__.from_bargmann(
-            [0], (math.inv(A[0]), -math.inv(A[0]) @ b[0], 1 / invert_this_c)
+        actual_inverse = self._from_attributes(
+            Bargmann(math.inv(A[0]), -math.inv(A[0]) @ b[0], 1 / invert_this_c),
+            self.wires,
+            self.name + "_inv",
```

Comment:
To confirm, this still uses the "shortcut" that we are using the Bargmann representation to store A,b,c's that don't necessarily correspond to the Bargmann representation?

Reply:
Yes, it's not an actual Bargmann representation (only on the input side). This and a few others will be updated when we have MultiRepresentation